src/components/SignUp.tsx

```diff
@@ -42,12 +44,23 @@ const SignUp = () => {
   >(MUTATION_KEYS.SIGN_UP);
   const [searchParams] = useSearchParams();
 
+  // used for resend email
+  const { mutate: signIn } = useMutation<unknown, unknown, { email: string }>(
```

Comment:
I don't think it's good to mix sign up and sign in in the same component (especially if the component is called sign up). What about assuming `resend button` is supposed to always send sign in, so the behavior can be defined in `SuccessContent`.